### PR TITLE
Fix: Save checkpoints more frequently and reset step counter for transfer learning

### LINKED ISSUE #1: Potential unintended bug in transfer learning phase?
Hi,

I've been reading your paper "World Model Agents with Change-Based Intrinsic Motivation" with great interest. Thanks a lot for sharing your work and the accompanying code.

While working to understand and reproduce your experiments, I came across a couple of implementation details that may have significantly impacted the transfer learning results. I wanted to share them in case they are helpful.

**1. Checkpoint saving might occur too infrequently.**

In [`train_eval.py`](https://github.com/Jazhyc/world-model-policy-transfer/blob/master/dreamerv3/embodied/run/train_eval.py#L16C3-L16C53), the checkpoint saving schedule is configured with `embodied.when.Clock(args.save_every)`. This triggers saves based on wall-clock time. For a `save_every` value of `1e5` (as in the [`crafter-pre-train` config](https://github.com/Jazhyc/world-model-policy-transfer/blob/master/dreamerv3/configs.yaml#L230)), a save occurs approximately every 27.8 hours.

On our machine, this unfortunately meant that no checkpoint is saved after the initial one, because both Crafter and MiniGrid runs finish in less than 27.8 hours. This led to mistakenly transferring an untrained model to the fine-tuning stage, where it then doesn't do any better than a model trained from scratch (since it is, in fact, a model trained from scratch). 

As far as I can tell, this could also have been problematic in your paper, since you e.g. reported a 12-hour pre-training run for Minigrid. For your longer 48-hour Crafter runs, it could still mean that the checkpoint used for transfer would be from an intermediate point in training, not from the end of pretraining. 

**2. The step counter is not reset for the transfer learning phase.**

The changes mentioned above helped to load the checkpoints correctly from the fully pretrained agents, but weren't enough to make the transfer learning work: I also noticed that when a [checkpoint is loaded for the transfer phase](https://github.com/Jazhyc/world-model-policy-transfer/blob/master/dreamerv3/embodied/run/train_eval.py#L124C3-L133C28), it restores the `step` counter's value from pre-training. This means the agent does not receive the full number of fine-tuning steps. For example, if the agent is pre-trained for 1M steps and fine-tuning is set to run *until* 1M steps, the agent will stop immediately without any fine-tuning, because the step counter is already at 1M.

Combined, these two issues likely mean the agent being transferred is not fully pre-trained, and it then receives far less fine-tuning than intended, which could explain why your CBET pretraining (Fig. 3, black curves in the plots on the right) did not seem to improve upon the extrinsic-only tabula rasa agent (Fig. 3, blue curves in the plots on the left) of your paper, even though CBET helped when directly added to the extrinsic reward in Crafter (Fig 3., black curves in the plots on the left, bottom row).

I have fixed this in my local version by setting `save_every` to `900` (every 15 minutes, as in the [original dreamerv3 repo](https://github.com/danijar/dreamerv3/blob/cdf570902b1eaba193cc8ef69426cd4edde1b0bc/dreamerv3/configs.yaml#L54)), as well as resetting the step counter before the transfer phase starts (`if args.transfer: step.value = 0`). If you think it would be helpful, I am happy to open a pull request with these fixes.

I hope this feedback is constructive. Thank you again for the insightful paper and for making your code available.

Best,
Freddy

### PR DESCRIPTION
**Pull request by freddyman2k**
Hi,

This pull request addresses a bug in the transfer learning phase and fixes #1.

The changes include:
1.  Updating the checkpoint saving frequency (from ~28 hours to 15 minutes) in `train_eval.py` to ensure a fully pre-trained model is saved.
2.  Resetting the step counter at the beginning of the transfer learning phase to allow for the intended number of additional training steps.

These fixes should ensure that the transfer learning experiments run as intended and produce more accurate results.

Best,
Freddy

### CODE REVIEW

**Comment by Jazhyc** [comment]
Thank you for the changes!
